tools/generate_report.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+
+#############################################################################
+##
+##  This file is part of GAP, a system for computational discrete algebra.
+##
+##  Copyright of GAP belongs to its developers, whose names are too numerous
+##  to list here. Please refer to the COPYRIGHT file for details.
+##
+##  SPDX-License-Identifier: GPL-2.0-or-later
+##
+
+# Run this script on main branch with data and gh-pages worktree
+
+"""
+This script compares the current test-status.json with a previous version,
+generates a main report.md along with
+    a test-status-diff.json,
+    a html-redirect, and
+    a badge.
+"""
+
+from utils import warning, error, symlink, string_to_bool
+
+import sys
+import os
+import json
+
+################################################################################
+# Arguments and Paths
+num_args = len(sys.argv)
+
+if num_args < 1 or num_args > 4:
+    error('Unknown number of arguments')
+
+# relative paths to report directories from ROOT
+ROOT = 'data/reports'
```

Comment:
Just to say, using all-uppercase identifiers is very unusual for python code.

Reply:
Yes, going from bash and python back and forth always messes me up :)